lib/tcpip/network_wrapper.cc

```diff
@@ -558,9 +558,9 @@ int network_socket_close(Timeout *t, SObj mallocCapability, SObj sealedSocket)
 	}
 	return with_sealed_socket(
 	  [=](SealedSocket *socket) {
-		  // Don't use a lock guard here, we don't want to release the lock if
-		  // it's been deallocated.  This must be released on all return paths
-		  // except the one for success.
+		  // Don't use a lock guard here, we don't want to release the
```

Comment:
We should probably add a `reset` method to the lock guard to swap the lock that is wraps (optionally with no lock) so that we can default to unlocking on all return paths but release lock ownership on the one where we've got rid of everything.